svc-aggregation/system/addaggregationsource.go

```diff
@@ -82,7 +83,31 @@ func (e *ExternalInterface) addAggregationSource(taskID, targetURI, reqBody stri
 	}
 
 	ipAddr := getKeyFromManagerAddress(addResourceRequest.ManagerAddress)
-
+	
+    systems, dberr := agmodel.GetAllSystems()
+	if dberr != nil {
+		errMsg := dberr.Error()
+		if errors.DBKeyNotFound == dberr.ErrNo() {
```

Comment:
this condition can be removed  when no bmc is present in odim still the add system needs to be continued